shared/ui/combobox/combobox.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { ChangeEvent, FC, MouseEventHandler, useState } from 'react';
+import { clsx } from 'clsx';
+
+import { Combobox as ComboboxUI } from '@headlessui/react';
+import * as ScrollArea from '@radix-ui/react-scroll-area';
+
+import style from './combobox.module.scss';
+
+import { Typography } from '..';
+
+import ArrowDown from '../../../assets/icons/arrow-down.svg?url';
+import Close from '../../../assets/icons/close.svg?url';
+
+import Image from 'next/image';
+
+type Option = {
+  label: string;
+  value: number | string;
+};
+
+export type ComboboxProps = {
+  disabled?: boolean;
+  errorMessage?: string;
+  inputValue: string;
+  label?: string;
+  name?: string;
+  onChange: (value: null | string) => void;
+  onClear?: () => void;
+  onInputChange: (value: string) => void;
+  options: Option[];
+  placeholder?: string;
+  showClearButton?: boolean;
+  value: string;
+};
+
+export const Combobox: FC<ComboboxProps> = props => {
+  const {
+    disabled,
+    errorMessage,
+    inputValue,
+    label,
+    name,
+    onChange,
+    onClear,
+    onInputChange,
+    options,
+    placeholder,
+    showClearButton = true,
+    value,
+  } = props;
+
+  const showError = !!errorMessage && errorMessage.length > 0;
+
+  const inputChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.currentTarget.value === '') {
+      onChange(null);
+    }
+    onInputChange(e.currentTarget.value);
+  };
+
+  const handleClearButtonClicked: MouseEventHandler<HTMLDivElement> = () => {
+    onInputChange('');
+    onChange(null);
+  };
+
+  const filteredOptions =
+    inputValue === ''
+      ? options
+      : options.filter(option => option.label.toLowerCase().includes(inputValue.toLowerCase()));
+
+  const getDisplayingValue = (value: number | string) =>
+    options?.find(option => option.value === value)?.label || '';
+
+  const s = {
+    container: style.container,
```

Comment:
What is the purpose of this object? Why not just pass style.container and others as `className` directly?

Reply:
I used this object to combine multiple classnames like s.input and s.options. I can delete some classNames, but is it gonna be ok to make it in different ways?

Reply:
You can remove duplicate classes from the `s` object and keep the ones that are computed. You can rename `s` to `computedClasses` or something like that to explicitly specify its purpose.

**Tip**: you can remove the `.input` class name altogether. Just insert its contents into the `.inutField` class name, e.g. like this:  
```css
.inputField {
  /* existing properties */

  &[data-headlessui-state='open'] {
      padding-right: 32px;
  
      &.hasClearButton {
        padding-right: 52px;
      }
    }
}
```